examples/exchange-files-in-browser/public/js/app.js

```diff
@@ -70,41 +74,25 @@ function createFileBlob (data, multihash) {
 }
 
 function getFile () {
-  const multihash = $multihashInput.value
+  const cid = $multihashInput.value
 
   $multihashInput.value = ''
 
   $errors.className = 'hidden'
 
-  if (!multihash) {
-    return console.log('no multihash was inserted')
-  }
+  if (!cid) { return console.log('no multihash was inserted') }
 
-  // files.get documentation
-  // https://github.com/ipfs/interface-ipfs-core/tree/master/API/files#get
-  node.files.get(multihash, (err, filesStream) => {
-    if (err) {
-      return onError(err)
-    }
+  node.files.get(cid, (err, files) => {
+    if (err) { return onError(err) }
 
-    filesStream.on('data', (file) => {
+    files.forEach((file) => {
       if (file.content) {
-        const buf = []
-        // buffer up all the data in the file
-        file.content.on('data', (data) => buf.push(data))
-
-        file.content.once('end', () => {
-          const listItem = createFileBlob(buf, multihash)
-
-          $filesList.insertBefore(listItem, $filesList.firstChild)
-        })
-
-        file.content.resume()
+        console.log('Fetched file:', cid, file.content.length)
+        // calling createFileBlob makes the Chrome go "Oh Snap"
+        // const listItem = createFileBlob(file.content, cid)
+        // $filesList.insertBefore(listItem, $filesList.firstChild)
```

Comment:
Ok, narrowed down to the createFileBlob func. Apparently it always makes the Chrome go "oh Snap" and in Firefox it creates a wrong file (3x the size)

@VictorBjelkholm thoughts on this one?